### PR TITLE
Adding PR template

### PR DESCRIPTION
**Pull request by fabiocav**
This change adds a PR template requesting required information and providing a checklist to ensure tasks associated with the changes are not missed.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/pull_request.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+resolves #issue_for_this_pr
+
+### Pull request checklist
+
+* [ ] My changes **do not** require documentation changes
+    * [ ] Otherwise: Documentation issue linked to PR
+* [ ] My changes **should not** be added to the release notes for the next release
+    * [ ] Otherwise: I've added my notes to `release_notes.md`
+
```
Comment (pragnagopa) [inline .github/ISSUE_TEMPLATE/pull_request.md:16]: Reminder on fix needs to be ported to V2?

Comment (pragnagopa) [reply .github/ISSUE_TEMPLATE/pull_request.md:16]: Reminder on UnitTests and E2E tests either automated or manual.

**Review by pragnagopa — COMMENTED** [review]
Added couple of comments.Thanks.

File: .github/ISSUE_TEMPLATE/pull_request.md (hunk 1 above)
Comment (fabiocav) [reply .github/ISSUE_TEMPLATE/pull_request.md:16]: Great call out on the V2 backport.

Debated on the tests, but I think it would be good to add.

**Review by pragnagopa — APPROVED** [review]

**Pull request merged**